sdks/java/io/cassandra/src/main/java/org/apache/beam/sdk/io/cassandra/CassandraIO.java

```diff
@@ -131,16 +132,16 @@ private CassandraIO() {}
   @AutoValue
   public abstract static class Read<T> extends PTransform<PBegin, PCollection<T>> {
     @Nullable
-    abstract List<String> hosts();
+    abstract ValueProvider<String> hosts();
```

Comment:
This does not preserve the same type, how can we past multiple hosts in this case? (Note that I don't know how Lists are supported by ValueProviders, worth the check.

Reply:
That's true, as far as I can see ValueProviders is used only for basic types but will check that.

Reply:
Re your question - you pass it as host1,host2,host3 - joined.

Reply:
Ok, I can [see](https://github.com/apache/beam/blob/b953645ed6db837d24284d7fe1fe091e7309f821/sdks/java/core/src/test/java/org/apache/beam/sdk/options/ValueProviderTest.java#L75) it is doable so I will check how it works with DataflowTemplates

Reply:
DataflowSDK doesn't handle it properly at this moment, there is [bug](https://github.com/GoogleCloudPlatform/DataflowJavaSDK/issues/632) so I would prefer to push it as is and in the meantime will try to investigate SDK project.

Reply:
I see, the problem is that pushing it with the string parameter implies a new signature that we must respect in the future, maybe we better just assume that the ValueProvider API will work with lists and update the PR consequently. Because the fact that Dataflow supports it is a different story from the fact that the API at least in principle does it.
Can you please fill a JIRA to report the ValueProvider issue in the Beam side so we can assign it to some googler.

Reply:
Ok, will rewrite this PR to support List. Bug is in DataflowSDK outside of apache.

Reply:
ok, done that.